spring-shell-core/src/main/java/org/springframework/shell/Shell.java

```diff
@@ -158,6 +143,7 @@ public Object evaluate(Input input) {
 
 		List<String> words = input.words();
 		if (command != null) {
+			Map<String, MethodTarget> methodTargets = commandRegistry.listCommands();
```

Comment:
A slight improvement would be to move this into a `methodTargets()` method (used in 3 places) and it may also be an easy area of extension to allow getting commands in a slightly different fashion.